### PR TITLE
Claim pending audit access during Login flow

### PR DESCRIPTION
**Pull request by timoballard**
Addresses #320 

Updates the backend Login.gov authentication flow so that:
- If a matching `User` record already exists, but no `LoginGovUser` record exists, a `LoginGovUser` record is created and associated with the existing `User` record
- Existing `Users` are looked up by finding all records where the email address is found in the LoginGov `all_emails` collection
  - If a single matching `User` is found, use it
  - If no matching `User` is found, create one
  - If multiple matching `Users` are found, use the one that matches the LoginGov primary email address
    - If none of them match the LoginGov primary email address, use the first one
- Find any outstanding `Access` records associated with an email address found in the `all_emails` collection
  - Link the `Access` record to the `User` ID

### CODE REVIEW

**Review by rachelschneiderman — APPROVED** [review]

File: backend/users/auth.py [hunk 1]
```diff
@@ -10,17 +10,59 @@
...
 
 class JWTUpsertAuthentication(JWTAuthentication):
+    def get_or_create_auth_user(self, email, all_emails):
+        # find all Users where email is in LoginGov all_emails
+        users = self.user_model.objects.filter(email__in=all_emails)
+
+        if users.count() > 1:
+            # more than one found, take the one that matches the LoginGov primary email
+            try:
+                user = users.get(email=email)
```
Comment (tadhg-ohiggins) [inline backend/users/auth.py:39]: Is it possible that there might be more than one user with the same email address?

File: backend/users/auth.py [hunk 2]
```diff
@@ -10,17 +10,59 @@
...
+                # none of the Users use the LoginGov primary email, take the first one
+                user = users[0]
+        elif users.count() == 1:
+            # found a single User match
+            user = users[0]
+        else:
+            # didn't find an existing User, create one
+
+            # generate a random password
+            password = "".join(
```
Comment (tadhg-ohiggins) [inline backend/users/auth.py:50]: Do we want to note here that this password is never going to be used, since our auth is all via login.gov?

Comment (timoballard) [reply backend/users/auth.py:51]: yes - done!

File: backend/users/auth.py (hunk 1 above)
Comment (timoballard) [reply backend/users/auth.py:39]: per our Slack discussion: uniqueness of `User` emails is not enforced at the database level, but this scenario should not be possible at the application level, given the way new `Users` are created. It is possible, though, if `Users` are created manually, either via direct database insertions or through Django admin. So, we explicitly catch and log that failure mode and ultimately reject the authentication attempt

**Review by tadhg-ohiggins — APPROVED** [review]

**Comment by tadhg-ohiggins** [comment]
Looks good to me, nice work!

**Pull request merged**